l1-contracts/contracts/upgrades/BaseZkSyncUpgrade.sol

```diff
@@ -199,10 +200,13 @@ abstract contract BaseZkSyncUpgrade is ZkSyncHyperchainBase {
 
         TransactionValidator.validateUpgradeTransaction(_l2ProtocolUpgradeTx);
 
+        // We expect that the upgrade transactions are only present when the minor version changes
+        (, uint32 minorVersion, ) = SemVer.unpackSemVer(_newProtocolVersion);
```

Comment:
nit: I would add two asserts in `BaseZkSyncUpgrade`
- check `majorVersion == 0`
- if upgrade changes anything except for verifier, check that `minorVersion` increases